processor/src/main/java/org/mapstruct/ap/internal/model/source/SourceMethod.java

```diff
@@ -183,9 +183,15 @@ public Builder setDefininingType(Type definingType) {
 
         public SourceMethod build() {
 
-            MappingOptions mappingOptions =
-                new MappingOptions( mappings, iterableMapping, mapMapping, beanMapping, valueMappings );
-
+            MappingOptions mappingOptions = null;
+            if ( (mappings != null && !mappings.isEmpty())
+                || iterableMapping != null
+                || mapMapping != null
+                || beanMapping != null
+                || valueMappings != null ) {
```

Comment:
Here when the `valueMappings` are not null they can be empty. What exactly we need to check here? 

If we don't generate `mappings`. Then `@InheritInverseConfuguration` will _only_ pick up annotated methods from the same mapper as well.